native/src/main/java/io/ballerina/stdlib/crypto/nativeimpl/Decrypt.java

```diff
@@ -100,4 +100,24 @@ public static Object decryptPgp(BArray cipherTextValue, BString privateKeyPath,
             return CryptoUtils.createError("Error occurred while PGP decrypt: " + e.getMessage());
         }
     }
+
+    public static Object decryptPgpAsFile(BString inputFilePath, BString privateKeyPath, BArray passphrase,
+                                          BString outputFilePath) {
+        byte[] passphraseInBytes = passphrase.getBytes();
+        byte[] privateKey;
+        try {
+            privateKey = Files.readAllBytes(Path.of(privateKeyPath.toString()));
+        } catch (IOException e) {
+            return CryptoUtils.createError("Error occurred while reading private key: " + e.getMessage());
+        }
+
+        try (InputStream keyStream = new ByteArrayInputStream(privateKey);
+            InputStream cipherTextStream = Files.newInputStream(Path.of(inputFilePath.toString()))) {
+            PgpDecryptionGenerator pgpDecryptionGenerator = new PgpDecryptionGenerator(keyStream, passphraseInBytes);
```

Comment:
Is this indentation correct (since the 2nd line is inside the try block)? 
```suggestion
            InputStream cipherTextStream = Files.newInputStream(Path.of(inputFilePath.toString()))) {
            PgpDecryptionGenerator pgpDecryptionGenerator = new PgpDecryptionGenerator(keyStream, passphraseInBytes);
```

Reply:
Will fix

Reply:
Fixed via [5ba88d7](https://github.com/ballerina-platform/module-ballerina-crypto/pull/567/commits/5ba88d7d21748a54d11ba4f7eda224128b2b288c)